src/VectorBundle.jl

```diff
@@ -0,0 +1,461 @@
+
+"""
+    VectorBundle{𝔽,TVS,TM,TVT}
```

Comment:
To me the whole generality of Fiber bundle is not yet so clear, so here saying that it is just a special case is maybe not helping a user if it does not help me?

I would first prefer a short description of what this actually is before noting that it is a special case of another type.

Reply:
Good point. What do you think about re-using old `VectorBundle` docs here? Or do you want more details?

Fiber bundles is not something commonly encountered in numerical applications but I think it's a good abstraction with a lot of supporting theory: unit tangent bundle, principal bundles, normal bundles, etc, with rich enough theory of metrics and connection forms that useful stuff can be done in many cases.

Reply:
Then let's mention a bit more details here, I would prefer either a short description of what that is, and/or a reference?

Reply:
I will work on the docs over the next few days. I will try to make it comprehensible.

Reply:
I am happy to help, for the ones I noted here, I just felt a bit lost.

Thanks for working on those, I think this is a great opportunity to extend them a bit.